### PR TITLE
refactor: move alignment code to linopy/alignment.py; restructure tests

### PR DESCRIPTION
**Pull request by FBumann**
Follow-up requested by @FabianHofmann in #732:

> move all broadcasting and alignment code from `common.py` to a new module `alignment.py`; move all related tests to `test_alignment.py`, make more use of test classes to structure them better

## Commit 1 — the module move (pure, no behavior change)

`linopy/alignment.py` owns the seam between user input and linopy's labelled arrays:

| Group | Functions |
|---|---|
| coords parsing | `_coords_to_dict`, `_as_index`, `_as_multiindex` |
| conversion | `get_from_iterable`, `pandas_to_dataarray`, `numpy_to_dataarray`, `_named_pandas_to_dataarray`, `fill_missing_coords`, `as_dataarray` |
| MI projection | `_LevelProjection`, `_project_onto_multiindex_levels`, `_warn_implicit_projections` (with `TODO(#738)`) |
| broadcasting | `_broadcast_to_coords`, `broadcast_to_coords` |
| validation | `validate_alignment` |
| symmetric | `align` |

`common.py` keeps the general utilities (formatting, label indexes, polars helpers, decorators). Hard move, no re-exports (nothing outside linopy imports these from `linopy.common` — checked PyPSA). The split is clean: nothing remaining in `common.py` depends on anything moved.

## Commit 2 — the test restructure

`test/test_alignment.py`, one class per concept:

```
TestAsDataarrayFrom{Pandas,Numpy,Scalar,DataArray}, TestAsDataarrayMultiIndexCoords
TestCoordsToDict            # the coords-entry naming rules
TestBroadcastToCoords       # strict=False mechanics
TestMultiIndexProjection    # projection values, deprecation warnings, gaps
TestStrictMode              # strict=True contract
TestValidateAlignment
TestAlign
```

- **Fixtures**: `mi_index` / `mi_coords` / `by_level1` replace the MultiIndex setup that was repeated inline in ~10 tests.
- **Parametrization**: the pandas dims-naming rules (12 near-identical tests → 2 parametrized tables) and the numpy labeling rules (9 → 1 table + 2 edge cases).
- `test_common.py` keeps the utility tests (`iterate_slices`, polars, formatting, …).

## v1 readiness (#717)

`TestMultiIndexProjection` collects every test that forks under the v1 semantics (deprecation warnings → raises). When #717's autouse `semantics` fixture lands, these get `@pytest.mark.legacy` and the v1 counterparts slot into the same class — no reorganization needed. Markers themselves are deliberately **not** added here (they belong to #717, where they have meaning).

## Verification — coverage measured, not just asserted

| | master (before) | this PR |
|---|---|---|
| `common.py` (monolith) | 787 stmts, 56 missed → 91% | — |
| `alignment.py` | — | 286 stmts, **2 missed → 99%** |
| `common.py` (utilities) | — | 516 stmts, 40 missed → 90% |
| combined missed lines | 56 | **42** |
| full suite | 3202 passed | **3210 passed** (8 new edge-case tests) |

The restructure preserved coverage exactly (the same missed lines, split across files), and the third commit then closed most of the remaining gaps: bare-string dims, 0-d arrays, partially-named MI levels, gap detection with extra dims, gap-error truncation, and the previously **zero-coverage MultiIndex round-trip** in `coords_to_dataset_vars` / `coords_from_dataset` (CSRConstraint serialization). The 2 remaining uncovered lines in `alignment.py` are defensive branches for inputs outside the `DimsLike` type contract.

- mypy clean, pre-commit clean.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Review by FabianHofmann — APPROVED** [review]
lovely!

**Comment by FBumann** [comment]
@FabianHofmann I need to convince myself that these changes are thorough and complete. Ill mark it as ready later

File: test/test_alignment.py [hunk 1]
```diff
@@ -0,0 +1,982 @@
...
+        assert list(da.coords["dim_0"].values) == ["a", "b"]
+        assert list(da.coords["dim_2"].values) == ["A", "B"]
+
+
+# ---------------------------------------------------------------------------
+# Implicit MultiIndex-level projection — the legacy/v1 fork point
+# ---------------------------------------------------------------------------
+
+
+class TestMultiIndexProjection:
```
Comment (FabianHofmann) [inline test/test_alignment.py]:
```suggestion
class TestBroadcastToCoordsMultiIndexProjection:
```

File: test/test_alignment.py [hunk 2]
```diff
@@ -0,0 +1,982 @@
...
+                sparse_unnamed, coords, dims=["snapshot"], label="lower bound"
+            )
+
+
+# ---------------------------------------------------------------------------
+# broadcast_to_coords(strict=True) — the contract
+# ---------------------------------------------------------------------------
+
+
+class TestStrictMode:
```
Comment (FabianHofmann) [inline test/test_alignment.py]:
```suggestion
class TestBroadcastToCoordsStrictMode:
```

**Review by FabianHofmann — APPROVED** [review]
only these two, feel free to merge after
